### PR TITLE
Improves kubernetes provider examples documentation

### PR DESCRIPTION
**Pull request by rodrmartinez**
Solves #850

### CODE REVIEW

File: docs/provider-kubernetes.md [hunk 1]
```diff
@@ -92,4 +85,69 @@ spec:
...
+kind: SecretStore
+metadata:
+  name: example
+spec:
+  provider:
+      kubernetes: 
+        # If not remoteNamesapce is provided, default     namespace is used
+        remoteNamespace: remote-namespace
+        server: 
+          url:  https://127.0.0.1:36473
```
Comment (gusfcarvalho) [inline docs/provider-kubernetes.md:123]: I think we can use a dns proxy, like `https://remote.kubernetes.api-server.address`

**Comment by gusfcarvalho** [comment]
@ADustyOldMuffin Do you mind checking if the new redact helped with k8s provider understanding?

**Comment by ADustyOldMuffin** [comment]
Looks excellent!

**Comment by gusfcarvalho** [comment]
thanks @rodrmartinez for the update :rocket:

**Pull request merged**